### ISSUE TITLE
Cannot add / remove / toggle element class inside Vue directive?

### ISSUE THREAD

**Issue by 7ammer** [vuejs/vue#5368]
### Version
2.2.6

### Reproduction link
[https://jsfiddle.net/hLga2jxq/](https://jsfiddle.net/hLga2jxq/)

### Steps to reproduce
Click the button in jsfiddle

### What is expected?
"Hello! :D" text should show and hide on click of button

### What is actually happening?
console shows classes where toggled but nothing visually happens

---
This is an example without the directive just to show what should be happening: https://jsfiddle.net/hxyv40ra/

**Comment by 7ammer** [comment]
So I tried this from another angle and also made the 'hide-name' attribute a directive as well, then on click I emitted the 'uniqueID' which 'hide-name' directive picked up.

I'm still not sure why Vue is not visually updating the browser though. I'm guessing it must have something to do with the 'virtual-dom'? In which case this might not be a bug.

Demo: https://jsfiddle.net/hLga2jxq/3/ (if anyone is interested)
